### ISSUE TITLE
Question: Get notice when command is executed

### ISSUE THREAD

**Issue by AlexBacich** [fluttercommunity/rx_command#35]
RefreshIndicator widget requires parameter Future<void> in order to function (finish circling value received). 

I'm wondering if there is a way to make RxCommand friend with RefreshIndicator. 

Problem is that once I call execute() for my command I get void result. 

What I want to do is call command and return Future with null emitted once command finishes with data or error.

**Comment by escamoteur** [comment]
what about using isExecuting.where(x=>0==false).take(1)? do that in a getter so that it is executed everytime?

**Comment by AlexBacich** [comment]
Hi @escamoteur , thanks for the answer. Here's solution I've came up so far. I'm sure it can be optimized but don't yet no ideas

```
Future<void> getHistory() async {
      getHistoryCommand.execute();
      return getHistoryCommand.isExecuting.where((b) => b == false).take(2).last;
    }
```

Even if I add delay after  getHistoryCommand.execute() it still fires isExecuting = false first. So I have to put  take(2) instead of take(1)

**Comment by escamoteur** [comment]
Ah, that is interesting. Not sure if I should count this as an error.
May I propose:

getHistoryCommand.isExecuting.distinctUnique.where((b) => b == false).take(1).last

Then it won't break in case I deide to remove the initial emitting of false.

Great that this works for you.

**Comment by AlexBacich** [comment]
I guess supposed behavior is to emit initially false only if command isn't executing. 
Anyway, what you are proposing is excellent idea. 
If I'm not the only one who needs this maybe helper function will be nice to have. Like 

> Future<void> executeAsFuture([param])  {
>      command.execute([param]);
>     getHistoryCommand.isExecuting.distinctUnique.where((b) => b == false).take(1).last;
> }

Will do in my project as extension anyway.

**Comment by escamoteur** [comment]
Good idea!

**Comment by AlexBacich** [comment]
Hey @escamoteur , just another observation in regards of commands flow. 

If I run command BEFORE widget if built for the first time - there are problems with creating RxLoader.
Basically it never receives state 'isExecuting = true' so not showing loader. Only after command is finished it receives first state. 

As a workaround solution - command should be launched in init state after widget is built once. Like this: 
WidgetsBinding.instance.addPostFrameCallback((_) => bloc.initialCommand());

Guess there's nothing to do here since moving all streams to behavior subjects would be an overkill. Or not?

**Comment by escamoteur** [comment]
Did you set the parameter `emitsLastValueToNewSubscriptions` to true? Then the Streams are BehaviourSubjects

**Comment by AlexBacich** [comment]
ah, you already did it ) It works, thanks!
